flink-table/flink-table-planner/src/test/java/org/apache/flink/table/sqlexec/SqlToOperationConverterTest.java

```diff
@@ -163,6 +169,229 @@ public void testSqlInsertWithStaticPartition() {
+			DataTypes.TIMESTAMP(3),
+			DataTypes.TIMESTAMP(3),
+			DataTypes.ARRAY(DataTypes.INT().notNull()),
+			DataTypes.ARRAY(DataTypes.INT()),
+			DataTypes.ARRAY(DataTypes.INT().notNull()),
+			DataTypes.ARRAY(DataTypes.INT()).notNull(),
+			DataTypes.MULTISET(DataTypes.INT().notNull()),
+			DataTypes.MULTISET(DataTypes.INT()),
+			DataTypes.MULTISET(DataTypes.INT().notNull()),
+			DataTypes.MULTISET(DataTypes.INT()).notNull(),
+			DataTypes.MAP(DataTypes.BIGINT(), DataTypes.BOOLEAN()),
+			// expect to be ROW<`f0` INT NOT NULL, `f1` BOOLEAN>, nullable attr is lost.
+			DataTypes.ROW(DataTypes.FIELD("f0", DataTypes.INT()),
+				DataTypes.FIELD("f1", DataTypes.BOOLEAN())),
+			// expect to be ROW<`f0` INT NOT NULL, `f1` BOOLEAN>, nullable attr is lost.
+			DataTypes.ROW(DataTypes.FIELD("f0", DataTypes.INT()),
+				DataTypes.FIELD("f1", DataTypes.BOOLEAN())),
+			DataTypes.ROW(DataTypes.FIELD("f0", DataTypes.INT())),
+			DataTypes.ROW(DataTypes.FIELD("f0", DataTypes.INT())),
+			DataTypes.ROW(),
+			DataTypes.ROW(),
+			// should be ROW<`f0` INT NOT NULL '...', `f1` BOOLEAN '...'>, the nullable attr
+			// and comments are both lost
+			DataTypes.ROW(DataTypes.FIELD("f0", DataTypes.INT()),
+				DataTypes.FIELD("f1", DataTypes.BOOLEAN()))
+		}, schema.getFieldDataTypes());
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType1() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: TIMESTAMP_WITH_LOCAL_TIME_ZONE");
+		final String sql = "create table t1(\n" +
+			"  f41 ARRAY<TIMESTAMP(3) WITH LOCAL TIME ZONE>)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType2() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: TIMESTAMP_WITH_LOCAL_TIME_ZONE");
+		final String sql = "create table t1(\n" +
+			"  f40 TIMESTAMP(3) WITH LOCAL TIME ZONE)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType3() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: TIMESTAMP_WITH_LOCAL_TIME_ZONE");
+		final String sql = "create table t1(\n" +
+			"  f39 TIMESTAMP WITH LOCAL TIME ZONE)";
+		final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+		SqlNode node = planner.parse(sql);
+		assert node instanceof SqlCreateTable;
+		SqlToOperationConverter.convert(planner, node);
+	}
+
+	@Test
+	public void testCreateTableWithUnSupportedDataType4() {
+		expectedEx.expect(TableException.class);
+		expectedEx.expectMessage("Type is not supported: VARBINARY");
```

Comment:
Why the exception message is not "BYTES"?

Reply:
I think it is because `BYTES` is recognized as `VARBINARY` internally, so there is no way to deduce which one the `VARBINARY` comes from.

Reply:
if the exception is generated with `LogicalType#asSummaryString` `BYTES` should be the representation

Reply:
No, the exception comes from Flink.